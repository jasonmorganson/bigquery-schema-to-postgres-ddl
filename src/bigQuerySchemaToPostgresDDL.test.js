const env = process.env
const argv = process.argv

beforeAll(() => {
    process.env = {
        ...env,
        SCHEMA: 'SCHEMA',
        TABLE: 'TABLE'
    }
    const [exec, path] = process.argv
    process.argv = [exec, path]
})

test.each`
    type           | json
    ${'BAD'}       | ${[{ type: 'FOO', name: 'BAR' }]}
    ${'STRING'}    | ${[{ type: 'STRING', name: 'STRING' }]}
    ${'INTEGER'}   | ${[{ type: 'INTEGER', name: 'INTEGER' }]}
    ${'NUMERIC'}   | ${[{ type: 'NUMERIC', name: 'NUMERIC' }]}
    ${'FLOAT'}     | ${[{ type: 'FLOAT', name: 'FLOAT' }]}
    ${'GEOGRAPHY'} | ${[{ type: 'GEOGRAPHY', name: 'GEOGRAPHY' }]}
    ${'BOOLEAN'}   | ${[{ type: 'BOOLEAN', name: 'BOOLEAN' }]}
    ${'DATE'}      | ${[{ type: 'DATE', name: 'DATE' }]}
    ${'TIMESTAMP'} | ${[{ type: 'TIMESTAMP', name: 'TIMESTAMP' }]}
`('Converting $type matches snapshot', ({ json }) => {
    const bigQuerySchemaToPostgresDDL = require('./bigQuerySchemaToPostgresDDL')
    const ddl = bigQuerySchemaToPostgresDDL(json)
    expect(ddl).toMatchSnapshot()
})

afterAll(() => {
    process.env = env
    process.argv = argv
})
