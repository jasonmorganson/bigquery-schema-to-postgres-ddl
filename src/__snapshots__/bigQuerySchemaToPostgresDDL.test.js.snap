// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Converting BAD matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" ()"`;

exports[`Converting BOOLEAN matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"BOOLEAN\\" boolean)"`;

exports[`Converting DATE matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"DATE\\" timestamptz)"`;

exports[`Converting FLOAT matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"FLOAT\\" FLOAT)"`;

exports[`Converting GEOGRAPHY matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"GEOGRAPHY\\" public.GEOGRAPHY)"`;

exports[`Converting INTEGER matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"INTEGER\\" integer)"`;

exports[`Converting NUMERIC matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"NUMERIC\\" NUMERIC)"`;

exports[`Converting STRING matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"STRING\\" text)"`;

exports[`Converting TIMESTAMP matches snapshot 1`] = `"create table \\"SCHEMA\\".\\"TABLE\\" (\\"TIMESTAMP\\" timestamptz)"`;
